src/primitives.rs

```diff
@@ -335,6 +337,53 @@ where
     }
 }
 
+impl<T> Versionize for VecDeque<T>
+where
+    T: Versionize,
+{
+    #[inline]
+    fn serialize<W: std::io::Write>(
+        &self,
+        mut writer: &mut W,
+        version_map: &VersionMap,
+        app_version: u16,
+    ) -> VersionizeResult<()> {
+        // Serialize in the same fashion as bincode:
+        // Write len.
+        bincode::serialize_into(&mut writer, &self.len())
+            .map_err(|ref err| VersionizeError::Serialize(format!("{:?}", err)))?;
+        // Walk the vecdeque and write each elemenet.
+        for element in self {
+            element
+                .serialize(writer, version_map, app_version)
+                .map_err(|ref err| VersionizeError::Serialize(format!("{:?}", err)))?;
+        }
+        Ok(())
+    }
+
+    #[inline]
+    fn deserialize<R: std::io::Read>(
+        mut reader: &mut R,
+        version_map: &VersionMap,
+        app_version: u16,
+    ) -> VersionizeResult<Self> {
+        let mut v = VecDeque::new();
+        let len: u64 = bincode::deserialize_from(&mut reader)
```

Comment:
nit-ish:

Serialized `len()` is `usize`: https://doc.rust-lang.org/std/collections/struct.VecDeque.html#method.len

Let's use the same thing here just in case `u64` differs in size from `usize`:
```suggestion
        let len: usize = bincode::deserialize_from(&mut reader)
```